validator_worker/src/sentry_interface.rs

```diff
@@ -1,143 +1,265 @@
+            })
+            .collect();
 
+        for validator in self.channel.spec.validators.into_iter() {
+            let auth_token = self.adapter.get_auth(&validator).unwrap();
+            match propagate_to(
+                &auth_token,
+                self.config.propagation_timeout,
+                &validator,
+                &serialised_messages,
+            ) {
+                Ok(_) => return,
+                Err(e) => handle_http_error(e, &validator.url),
+            }
+        }
     }
 
-    async pub fn get_latest_msg(from: &str, type: &str) -> impl Future<Output = Result<Vec<Channel>, Error>>  {
+    pub async fn get_latest_msg(
+        &self,
+        from: String,
+        message_type: String,
+    ) -> Result<ValidatorMessageResponse, reqwest::Error> {
         let future = self
             .client
-            .get(format!("{}/validator-messages/{}/{}?limit=1", self.sentry_url, from, type))
+            .get(&format!(
+                "{}/validator-messages/{}/{}?limit=1",
+                self.sentry_url, from, message_type
+            ))
             .send()
-            .and_then(|mut res: Response| res.json::<ChannelAllResponse>());
+            .and_then(|mut res: Response| res.json::<ValidatorMessageResponse>())
+            .compat();
 
-        await!(future.compat())
+        await!(future)
     }
 
-    async pub fn get_our_latest_msg(type: &str) -> impl Future<Output = Result<Vec<Channel>, Error>>  {
-        let whoami = adapter.whoami();
-        self.get_latest_msg(whoami, type)
+    pub async fn get_our_latest_msg(
+        &self,
+        message_type: String,
+    ) -> Result<ValidatorMessageResponse, reqwest::Error> {
+        let whoami = self.adapter.whoami();
+        await!(self.get_latest_msg(whoami, message_type))
+    }
+
+    pub async fn get_last_approved(&self) -> Result<LastApprovedResponse, reqwest::Error> {
+        let future = self
+            .client
+            .get(&format!("{}/last-approved", self.sentry_url))
+            .send()
+            .and_then(|mut res: Response| res.json::<LastApprovedResponse>());
+
+        await!(future.compat())
     }
 
-    pub fn get_last_approved() -> impl Future<Output = Result<Vec<Channel>, Error>> {
+    pub async fn get_last_msgs(&self) -> Result<LastApprovedResponse, reqwest::Error> {
         let future = self
             .client
-            .get(format!("{}/last-approved", self.sentry_url))
+            .get(&format!(
+                "{}/last-approved?withHearbeat=true",
+                self.sentry_url
+            ))
             .send()
-            .and_then(|mut res: Response| res.json::<ChannelAllResponse>());
+            .and_then(|mut res: Response| res.json::<LastApprovedResponse>());
 
         await!(future.compat())
     }
 
-    pub fn get_last_msgs() {
-         let future = self
+    pub async fn get_event_aggregates(
+        &self,
+        after: Option<u32>,
```

Comment:
what's this after u32? Can you be more specific? I don't thin this is really clear for somebody reading the code.

Reply:
it's basically "get all event aggregates AFTER this point" 

but it looks like it should be a DateTime, and not optional (cause you can always pass a zero timedate, 1970)

Reply:
yeah made changed it to Datetime as its more descriptive